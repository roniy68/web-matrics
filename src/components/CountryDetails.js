import { useParams } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { fetchCountryDetails } from '../redux/countryDetails/countryDetailsSlice';
import styles from '../styles/CountryDetails.module.css';

const CountryDetails = () => {
  const { countryId } = useParams();
  const dispatch = useDispatch();

  const country = useSelector((state) => state.countryDetails);

  useEffect(() => {
    if (Object.keys(country).length === 0) {
      dispatch(fetchCountryDetails(countryId));
    }
  }, [country, dispatch, countryId]);

  return (
    <>
      {Object.keys(country).length !== 0 && (
      <div className={styles.countryDetails}>
        <img src={country.flagSvg} alt={country.flagAlt} className={styles.flag} />
        <h1 className={styles.name}>{country.name}</h1>
        <ul className={styles.details}>
          <li className={styles.detail}>
            <span className={styles.detailName}>Capital:</span>
            {' '}
            <span className={styles.detailValue}>{country.capital}</span>
          </li>
          <li className={styles.detail}>
            <span className={styles.detailName}>Population:</span>
            {' '}
            <span className={styles.detailValue}>{country.population}</span>
          </li>
          <li className={styles.detail}>
            <span className={styles.detailName}>Area:</span>
            {' '}
            <span className={styles.detailValue}>
              {country.area}
              {' '}
              kmÂ²
            </span>
          </li>
          <li className={styles.detail}>
            <span className={styles.detailName}>Timezones:</span>
            {' '}
            <span className={styles.detailValue}>
              {country.timezones.map((timezone) => (
                <span key={timezone} className={styles.timezone}>{timezone}</span>
              ))}
            </span>
          </li>
        </ul>
      </div>
      )}
    </>
  );
};

export default CountryDetails;
