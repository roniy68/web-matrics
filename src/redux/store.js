import countriesReducer from './countries/countriesSlice';
import countryDetailsSlice from './countryDetails/countryDetailsSlice';

const { configureStore } = require('@reduxjs/toolkit');

const store = configureStore({
  reducer: {
    countries: countriesReducer,
    countryDetails: countryDetailsSlice,
  },
});

export default store;
