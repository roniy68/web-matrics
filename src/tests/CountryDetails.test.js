import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/store';
import CountryDetails from '../components/CountryDetails';

describe('CountryDetails', () => {
  it('should render the component', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <CountryDetails />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
