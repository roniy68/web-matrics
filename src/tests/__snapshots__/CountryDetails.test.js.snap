// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryDetails should render the component 1`] = `null`;
